packages/assets-controllers/CHANGELOG.md

```diff
@@ -28,8 +61,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - These types were previously used in TokenRatesController to represent data returned from the CoinGecko API. There is no equivalent.
 
 ### Fixed
-- Fix TokenRatesController to prevent redundant token rate updates when tokens change ([#3647](https://github.com/MetaMask/core/pull/3647), [#3663](https://github.com/MetaMask/core/pull/3663))
+- Prevent `TokenRatesController` from making redundant token rate updates when tokens change ([#3647](https://github.com/MetaMask/core/pull/3647), [#3663](https://github.com/MetaMask/core/pull/3663))
   - Previously, token rates would be re-fetched for the globally selected network on all TokensController state changes, but now token rates are always performed for a deduplicated and normalized set of addresses, and changes to this set determine whether rates should be re-fetched.
+- Prevent redundant overlapping token rate updates in `TokenRatesController` ([#3635](https://github.com/MetaMask/core/pull/3635))
+- Fix `TokenRatesController` bug where the `contractExchangeRates` state would sometimes be stale after calling `updateExchangeRatesByChainId` ([#3624](https://github.com/MetaMask/core/pull/3624))
+- Make `TokenRatesController.updateExchangeRatesByChainId` respect `disabled` state ([#3596](https://github.com/MetaMask/core/pull/3596))
+- Fix error in `NftController` when attempt to get NFT information from on-chain fails, and ensure metadata always contains contract address and blank `name` field ([#3629](https://github.com/MetaMask/core/pull/3629))
```

Comment:
It looks like this PR also fixed a potential issue with fetching details for ERC 1155 tokens: https://github.com/MetaMask/core/pull/3629/files#diff-07569734c8b1c2861eafbf941f191ded03f70ea952d886387e87915ac85a7c47R240. Previously, tokenURI could be undefined, yet an attempt would be made to make a request anyway. Now that we have proper types for `tokenURI`, TypeScript knows it could be undefined, so we are forced to check for it instead of `tokenId`; but this also makes it so that we only make the request if we have a URL.

But maybe that's what "Fix error in `NftController` when attempt to get NFT information from on-chain fails" implies?

Reply:
I only meant to reference the uses of `getNftInformation` with this change entry, hence the reference to this affecting `watchNft` and `addNft` (the two methods that call `getNftInformation`).

I thought the `tokenId` => `tokenURI` change had no functional impact. I'll take a closer look.

Reply:
Good catch, agreed that this is also a bug fix. I'll try to work this in.

Reply:
Actually on second thought, I don't think this constitutes a functional change because the "fetch undefined" step is in a try/catch block with an empty catch. So it would have caused an error previously, but it was suppressed. Whereas now it's skipped. I think it's the same end result, just nicer code now.

Reply:
Alright!